packages/apollo-server-plugin-base/src/index.ts

```diff
@@ -15,9 +15,9 @@ export {
   WithRequired,
 };
 
-export interface ApolloServerPlugin {
+export interface ApolloServerPlugin<TContext = Record<string, any>> {
```

Comment:
What about constraining this generic to match our expected context shape of `Record<string, any>` using `extends` to prevent assigning peculiar types to it?  (In addition to providing the default value.)

```suggestion
export interface ApolloServerPlugin<TContext extends Record<string, any> = Record<string, any>> {
```

Reply:
I think this is ready to merge after this suggestion is committed! @jdmoody I'll wait until next week to hear from you, but afterwards I think @abernix and I will land this suggestion (and PR).

Reply:
Good idea 👍 Updated!